test/EVMHost.cpp

```diff
@@ -189,14 +189,17 @@ void EVMHost::transfer(evmc::MockedAccount& _sender, evmc::MockedAccount& _recip
 	_recipient.balance = convertToEVMC(u256(convertFromEVMC(_recipient.balance)) + _value);
 }
 
-void EVMHost::selfdestruct(const evmc::address& _addr, const evmc::address& _beneficiary) noexcept
+bool EVMHost::selfdestruct(const evmc::address& _addr, const evmc::address& _beneficiary) noexcept
 {
 	// TODO actual selfdestruct is even more complicated.
 
 	transfer(accounts[_addr], accounts[_beneficiary], convertFromEVMC(accounts[_addr].balance));
 	accounts.erase(_addr);
-	// Record self destructs
-	recorded_selfdestructs.push_back({_addr, _beneficiary});
+
+	// Record self destructs.
+	auto& beneficiaries = recorded_selfdestructs[_addr];
+	beneficiaries.emplace_back(_beneficiary);
+	return beneficiaries.size() == 1;
```

Comment:
You can reuse MockedHost.

```suggestion
	return MockedHost::selfdestruct(_addr, _beneficiary);
```